apps/web/src/services/dashboards/shared/dashboard-widget-input-form/dashboard-widget-options-form/DashboardCostWidgetValueOptionDropdown.vue

```diff
@@ -1,55 +1,203 @@
 <script setup lang="ts">
-import { PIconButton, PSelectDropdown } from '@spaceone/design-system';
+import {
+    reactive, watch,
+} from 'vue';
+
+import { PIconButton, PSelectDropdown, PBadge } from '@spaceone/design-system';
 import type {
     AutocompleteHandler,
     SelectDropdownMenuItem,
 } from '@spaceone/design-system/types/inputs/dropdown/select-dropdown/type';
 
+import { SpaceConnector } from '@cloudforet/core-lib/space-connector';
+
+import ErrorHandler from '@/common/composables/error/errorHandler';
+
+import { useWidgetFormStore } from '@/services/dashboards/shared/dashboard-widget-input-form/widget-form-store';
+
 interface Props {
+    optionKey?: string;
     selected?: SelectDropdownMenuItem[];
-    menuHandlers?: AutocompleteHandler[];
     invalid?: boolean;
 }
 
 const props = withDefaults(defineProps<Props>(), {
+    optionKey: '',
     selected: undefined,
-    menuHandlers: undefined,
     invalid: false,
 });
 
 const emit = defineEmits<{(e: 'update:selected', selected: SelectDropdownMenuItem[]): void;
     (e: 'delete'): void;
 }>();
 
-const handleUpdateSelected = (selected: SelectDropdownMenuItem[]) => {
-    emit('update:selected', selected);
+const widgetFormStore = useWidgetFormStore();
+const widgetFormState = widgetFormStore.$state;
+
+const state = reactive({
+    visibleMenu: false,
+    reloadOnMenuHandlerUpdate: false,
+    selectedKey: {} as SelectDropdownMenuItem,
+    valueTarget: props.optionKey === 'cost_tag_value' ? 'tags' : 'additional_info',
+    selectedItems: [] as SelectDropdownMenuItem[],
+});
+
+const variableModelKeys = {
+    cost_tag_value: 'cost_tag_keys',
+    cost_additional_info_value: 'cost_additional_info_keys',
+} as const;
+
+const menuHandler: AutocompleteHandler = async () => {
```

Comment:
issue: arguments such as `inputText` must be passed and must be used at api param. :)